src/System.Private.CoreLib/shared/System/Runtime/CompilerServices/AsyncMethodBuilder.cs

```diff
@@ -915,16 +915,16 @@ internal static Task<TResult> GetTaskForResult(TResult result)
                 }
                 // For other known value types, we only special-case 0 / default(TResult).
                 else if (
-                    (typeof(TResult) == typeof(uint) && default == (uint)(object?)result) ||
-                    (typeof(TResult) == typeof(byte) && default(byte) == (byte)(object?)result) ||
-                    (typeof(TResult) == typeof(sbyte) && default(sbyte) == (sbyte)(object?)result) ||
-                    (typeof(TResult) == typeof(char) && default(char) == (char)(object?)result) ||
-                    (typeof(TResult) == typeof(long) && default == (long)(object?)result) ||
-                    (typeof(TResult) == typeof(ulong) && default == (ulong)(object?)result) ||
-                    (typeof(TResult) == typeof(short) && default(short) == (short)(object?)result) ||
-                    (typeof(TResult) == typeof(ushort) && default(ushort) == (ushort)(object?)result) ||
-                    (typeof(TResult) == typeof(IntPtr) && default == (IntPtr)(object?)result) ||
-                    (typeof(TResult) == typeof(UIntPtr) && default == (UIntPtr)(object?)result))
+                    (typeof(TResult) == typeof(uint) && default == (uint)(object?)result!) ||
+                    (typeof(TResult) == typeof(byte) && default(byte) == (byte)(object?)result!) ||
+                    (typeof(TResult) == typeof(sbyte) && default(sbyte) == (sbyte)(object?)result!) ||
+                    (typeof(TResult) == typeof(char) && default(char) == (char)(object?)result!) ||
+                    (typeof(TResult) == typeof(long) && default == (long)(object?)result!) ||
+                    (typeof(TResult) == typeof(ulong) && default == (ulong)(object?)result!) ||
+                    (typeof(TResult) == typeof(short) && default(short) == (short)(object?)result!) ||
+                    (typeof(TResult) == typeof(ushort) && default(ushort) == (ushort)(object?)result!) ||
+                    (typeof(TResult) == typeof(IntPtr) && default == (IntPtr)(object?)result!) ||
+                    (typeof(TResult) == typeof(UIntPtr) && default == (UIntPtr)(object?)result!)) // TODO-NULLABLE: https://github.com/dotnet/roslyn/issues/34976
```

Comment:
Can we remove the question marks from object? now that we're adding the !s?

Reply:
Actually yes... would it makes sense to change the typeof == typeof to use `is` instead? Or that would change IL?

Reply:
> Actually yes... would it makes sense to change the typeof == typeof to use is instead? Or that would change IL?

That would definitely change the IL (and resulting asm) in a bad way.  Please don't do that :)

Reply:
😄 🤦‍♂️ ok.

Reply:
Fixed.